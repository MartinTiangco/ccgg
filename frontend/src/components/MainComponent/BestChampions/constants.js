const RANKED = "ranked";
const NORMAL = "normal";
const OTHERS = "others";

export default { RANKED, NORMAL, OTHERS };
