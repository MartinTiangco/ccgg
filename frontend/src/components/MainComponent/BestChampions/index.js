import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import {
  AppBar,
  Card,
  Divider,
  Grid,
  Tab,
  Tabs,
  Typography,
} from "@material-ui/core";
import ErrorOutlineIcon from "@material-ui/icons/ErrorOutline";
import gameTypes from "./constants";

const ddragonChamps = `http://ddragon.leagueoflegends.com/cdn/11.9.1/img/champion/`;
const { RANKED, NORMAL, OTHERS } = gameTypes;
const maxNumChampionsShown = 8;

const useStyles = makeStyles((theme) => ({
  img: {
    width: "40px",
    height: "40px",
    marginRight: "5px",
    borderRadius: "5px",
  },
  firstContainer: {
    paddingTop: "5px",
  },
  divider: {
    width: "100%",
  },
  card: {
    padding: theme.spacing(2),
  },
  tab: {
    fontSize: "12px",
    minWidth: "33%",
  },
  noDataTab: {
    height: "150px",
  },
  championText: {
    color: "#ECF0F1",
    fontWeight: "bold",
    fontSize: "1rem",
  },
  bottomText: {
    fontSize: "12px",
    color: theme.palette.text.secondary,
  },
  kdaWinrateText: {
    fontSize: "14px",
  },
}));

const TabPanel = (props) => {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`tabpanel-${index}`}
      style={{ width: "100%" }}
      {...other}
    >
      {value === index && <>{children}</>}
    </div>
  );
};

const BestChampions = ({ championStats }) => {
  const classes = useStyles();
  const [value, setValue] = React.useState(0);

  const handleChange = (_, newValue) => {
    setValue(newValue);
  };

  const getDisplay = ({ champions }, queueTypeConstant) => {
    // if no data found
    if (champions.length === 0) {
      return (
        <Grid
          key={`${queueTypeConstant}-not-found`}
          className={`${classes.firstContainer} ${classes.noDataTab}`}
          item
          container
          direction="column"
          justify="center"
          alignItems="center"
        >
          <ErrorOutlineIcon />
          No data found!
        </Grid>
      );
    }

    return champions.slice(0, maxNumChampionsShown).map((entry, index) => {
      const { champion, csTotal, kills, deaths, assists, wr, games } = entry;

      const displayCs = `Avg. CS ${csTotal}`;
      const kda = `${
        Math.round(((kills + assists) / deaths + Number.EPSILON) * 100) / 100
      }`;

      // Wukong's name in the Riot API is annoyingly named MonkeyKing, hence we have to consider this edge case
      const championName = champion === "MonkeyKing" ? "Wukong" : champion;

      // let outerGridClassNames = classes.outerGrid;
      const outerGridClassNames = index > 0 ? classes.firstContainer : "";

      return (
        <div key={`${queueTypeConstant}-${championName}`}>
          <Grid
            className={outerGridClassNames}
            item
            container
            direction="row"
            style={{ padding: "5px" }}
          >
            <Grid item container direction="column" xs={2}>
              <img
                className={classes.img}
                src={`${ddragonChamps}${champion}.png`}
                alt={championName}
              />
            </Grid>

            <Grid item container direction="column" xs={4}>
              <Typography className={classes.championText} variant="h5">
                {championName}
              </Typography>
              <Typography className={classes.bottomText}>
                {displayCs}
              </Typography>
            </Grid>
            <Grid item container direction="column" xs={3}>
              <Typography
                className={classes.kdaWinrateText}
              >{`${kda}:1 KDA`}</Typography>
              <Typography
                className={classes.bottomText}
              >{`${kills}/${deaths}/${assists}`}</Typography>
            </Grid>
            <Grid item container direction="column" xs={3}>
              <Typography
                className={classes.kdaWinrateText}
              >{`${wr}%`}</Typography>
              <Typography
                className={classes.bottomText}
              >{`${games} games`}</Typography>
            </Grid>
          </Grid>
          <Divider className={classes.divider} />
        </div>
      );
    });
  };

  const rankedStats = getDisplay(
    championStats.find(({ type }) => type === RANKED),
    RANKED
  );
  const normalStats = getDisplay(
    championStats.find(({ type }) => type === NORMAL),
    NORMAL
  );
  const otherStats = getDisplay(
    championStats.find(({ type }) => type === OTHERS),
    OTHERS
  );

  return (
    <>
      <AppBar position="relative">
        <Tabs value={value} onChange={handleChange}>
          <Tab className={classes.tab} label="Ranked" />
          <Tab className={classes.tab} label="Normal" />
          <Tab className={classes.tab} label="Other" />
        </Tabs>
      </AppBar>

      <Card className={classes.card} raised>
        <Grid container>
          <TabPanel value={value} index={0}>
            {rankedStats}
          </TabPanel>
          <TabPanel value={value} index={1}>
            {normalStats}
          </TabPanel>
          <TabPanel value={value} index={2}>
            {otherStats}
          </TabPanel>
        </Grid>
      </Card>
    </>
  );
};

export default BestChampions;
