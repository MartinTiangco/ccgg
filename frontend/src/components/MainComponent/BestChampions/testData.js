// dummy data
const championStats = [
  {
    type: "ranked",
    champions: [
      {
        champion: "Ekko",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "Syndra",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "Renekton",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "Annie",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "Xerath",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "Orianna",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "Galio",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "Garen",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "Viktor", // this champ shouldn't be shown because there are already max number of champions shown
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
    ],
  },
  {
    type: "normal",
    champions: [
      {
        champion: "Lucian",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "MonkeyKing",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
    ],
  },
  {
    type: "others",
    champions: [],
  },
];

export default championStats;
