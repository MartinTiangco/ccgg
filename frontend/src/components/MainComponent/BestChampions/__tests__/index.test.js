import renderer from "react-test-renderer";
import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import BestChampions from "../index";

const testData = [
  {
    type: "ranked",
    champions: [
      {
        champion: "Ekko",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "Syndra",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "Renekton",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "Annie",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "Xerath",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "Orianna",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "Galio",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "Garen",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "Viktor", // this champ shouldn't be shown because there are already max number of champions shown
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
    ],
  },
  {
    type: "normal",
    champions: [
      {
        champion: "Lucian",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
      {
        champion: "MonkeyKing",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
    ],
  },
  {
    type: "others",
    champions: [],
  },
];

const testDataWithWukong = [
  {
    type: "ranked",
    champions: [
      {
        champion: "MonkeyKing",
        csTotal: 178.6,
        kills: 7,
        deaths: 3,
        assists: 17,
        wr: 58,
        games: 137,
      },
    ],
  },
  {
    type: "normal",
    champions: [],
  },
  {
    type: "others",
    champions: [],
  },
];

const testDataEmpty = [
  {
    type: "ranked",
    champions: [],
  },
  {
    type: "normal",
    champions: [],
  },
  {
    type: "others",
    champions: [],
  },
];

it("snapshot", () => {
  const tree = renderer
    .create(<BestChampions championStats={testData} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("MonkeyKing should be rendered to Wukong", () => {
  render(<BestChampions championStats={testDataWithWukong} />);

  expect(screen.getByText("Wukong")).toBeInTheDocument();
  expect(screen.queryByText("MonkeyKing")).toBeNull();
});

it("renders correctly if no champions are found", () => {
  render(<BestChampions championStats={testDataEmpty} />);

  expect(screen.getByText("No data found!")).toBeInTheDocument();
  fireEvent.click(screen.getByText("Normal"));
  expect(screen.getByText("No data found!")).toBeInTheDocument();
  fireEvent.click(screen.getByText("Other"));
  expect(screen.getByText("No data found!")).toBeInTheDocument();
});
