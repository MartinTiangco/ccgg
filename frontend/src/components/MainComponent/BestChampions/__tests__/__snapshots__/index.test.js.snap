// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiTabs-root"
    >
      <div
        className="MuiTabs-scroller MuiTabs-fixed"
        onScroll={[Function]}
        style={
          Object {
            "marginBottom": null,
            "overflow": "hidden",
          }
        }
      >
        <div
          className="MuiTabs-flexContainer"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-selected={true}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit makeStyles-tab-5 Mui-selected"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Ranked
            </span>
            <span
              className="PrivateTabIndicator-root-10 PrivateTabIndicator-colorSecondary-12 MuiTabs-indicator"
              style={Object {}}
            />
          </button>
          <button
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit makeStyles-tab-5"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Normal
            </span>
          </button>
          <button
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit makeStyles-tab-5"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Other
            </span>
          </button>
        </div>
      </div>
    </div>
  </header>,
  <div
    className="MuiPaper-root MuiCard-root makeStyles-card-4 MuiPaper-elevation8 MuiPaper-rounded"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        hidden={false}
        id="tabpanel-0"
        role="tabpanel"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div>
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item"
            style={
              Object {
                "padding": "5px",
              }
            }
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-2"
            >
              <img
                alt="Ekko"
                className="makeStyles-img-1"
                src="http://ddragon.leagueoflegends.com/cdn/11.9.1/img/champion/Ekko.png"
              />
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-4"
            >
              <h5
                className="MuiTypography-root makeStyles-championText-7 MuiTypography-h5"
              >
                Ekko
              </h5>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                Avg. CS 178.6
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root makeStyles-kdaWinrateText-9 MuiTypography-body1"
              >
                8:1 KDA
              </p>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                7/3/17
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root makeStyles-kdaWinrateText-9 MuiTypography-body1"
              >
                58%
              </p>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                137 games
              </p>
            </div>
          </div>
          <hr
            className="MuiDivider-root makeStyles-divider-3"
          />
        </div>
        <div>
          <div
            className="MuiGrid-root makeStyles-firstContainer-2 MuiGrid-container MuiGrid-item"
            style={
              Object {
                "padding": "5px",
              }
            }
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-2"
            >
              <img
                alt="Syndra"
                className="makeStyles-img-1"
                src="http://ddragon.leagueoflegends.com/cdn/11.9.1/img/champion/Syndra.png"
              />
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-4"
            >
              <h5
                className="MuiTypography-root makeStyles-championText-7 MuiTypography-h5"
              >
                Syndra
              </h5>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                Avg. CS 178.6
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root makeStyles-kdaWinrateText-9 MuiTypography-body1"
              >
                8:1 KDA
              </p>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                7/3/17
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root makeStyles-kdaWinrateText-9 MuiTypography-body1"
              >
                58%
              </p>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                137 games
              </p>
            </div>
          </div>
          <hr
            className="MuiDivider-root makeStyles-divider-3"
          />
        </div>
        <div>
          <div
            className="MuiGrid-root makeStyles-firstContainer-2 MuiGrid-container MuiGrid-item"
            style={
              Object {
                "padding": "5px",
              }
            }
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-2"
            >
              <img
                alt="Renekton"
                className="makeStyles-img-1"
                src="http://ddragon.leagueoflegends.com/cdn/11.9.1/img/champion/Renekton.png"
              />
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-4"
            >
              <h5
                className="MuiTypography-root makeStyles-championText-7 MuiTypography-h5"
              >
                Renekton
              </h5>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                Avg. CS 178.6
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root makeStyles-kdaWinrateText-9 MuiTypography-body1"
              >
                8:1 KDA
              </p>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                7/3/17
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root makeStyles-kdaWinrateText-9 MuiTypography-body1"
              >
                58%
              </p>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                137 games
              </p>
            </div>
          </div>
          <hr
            className="MuiDivider-root makeStyles-divider-3"
          />
        </div>
        <div>
          <div
            className="MuiGrid-root makeStyles-firstContainer-2 MuiGrid-container MuiGrid-item"
            style={
              Object {
                "padding": "5px",
              }
            }
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-2"
            >
              <img
                alt="Annie"
                className="makeStyles-img-1"
                src="http://ddragon.leagueoflegends.com/cdn/11.9.1/img/champion/Annie.png"
              />
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-4"
            >
              <h5
                className="MuiTypography-root makeStyles-championText-7 MuiTypography-h5"
              >
                Annie
              </h5>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                Avg. CS 178.6
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root makeStyles-kdaWinrateText-9 MuiTypography-body1"
              >
                8:1 KDA
              </p>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                7/3/17
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root makeStyles-kdaWinrateText-9 MuiTypography-body1"
              >
                58%
              </p>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                137 games
              </p>
            </div>
          </div>
          <hr
            className="MuiDivider-root makeStyles-divider-3"
          />
        </div>
        <div>
          <div
            className="MuiGrid-root makeStyles-firstContainer-2 MuiGrid-container MuiGrid-item"
            style={
              Object {
                "padding": "5px",
              }
            }
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-2"
            >
              <img
                alt="Xerath"
                className="makeStyles-img-1"
                src="http://ddragon.leagueoflegends.com/cdn/11.9.1/img/champion/Xerath.png"
              />
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-4"
            >
              <h5
                className="MuiTypography-root makeStyles-championText-7 MuiTypography-h5"
              >
                Xerath
              </h5>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                Avg. CS 178.6
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root makeStyles-kdaWinrateText-9 MuiTypography-body1"
              >
                8:1 KDA
              </p>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                7/3/17
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root makeStyles-kdaWinrateText-9 MuiTypography-body1"
              >
                58%
              </p>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                137 games
              </p>
            </div>
          </div>
          <hr
            className="MuiDivider-root makeStyles-divider-3"
          />
        </div>
        <div>
          <div
            className="MuiGrid-root makeStyles-firstContainer-2 MuiGrid-container MuiGrid-item"
            style={
              Object {
                "padding": "5px",
              }
            }
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-2"
            >
              <img
                alt="Orianna"
                className="makeStyles-img-1"
                src="http://ddragon.leagueoflegends.com/cdn/11.9.1/img/champion/Orianna.png"
              />
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-4"
            >
              <h5
                className="MuiTypography-root makeStyles-championText-7 MuiTypography-h5"
              >
                Orianna
              </h5>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                Avg. CS 178.6
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root makeStyles-kdaWinrateText-9 MuiTypography-body1"
              >
                8:1 KDA
              </p>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                7/3/17
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root makeStyles-kdaWinrateText-9 MuiTypography-body1"
              >
                58%
              </p>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                137 games
              </p>
            </div>
          </div>
          <hr
            className="MuiDivider-root makeStyles-divider-3"
          />
        </div>
        <div>
          <div
            className="MuiGrid-root makeStyles-firstContainer-2 MuiGrid-container MuiGrid-item"
            style={
              Object {
                "padding": "5px",
              }
            }
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-2"
            >
              <img
                alt="Galio"
                className="makeStyles-img-1"
                src="http://ddragon.leagueoflegends.com/cdn/11.9.1/img/champion/Galio.png"
              />
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-4"
            >
              <h5
                className="MuiTypography-root makeStyles-championText-7 MuiTypography-h5"
              >
                Galio
              </h5>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                Avg. CS 178.6
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root makeStyles-kdaWinrateText-9 MuiTypography-body1"
              >
                8:1 KDA
              </p>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                7/3/17
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root makeStyles-kdaWinrateText-9 MuiTypography-body1"
              >
                58%
              </p>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                137 games
              </p>
            </div>
          </div>
          <hr
            className="MuiDivider-root makeStyles-divider-3"
          />
        </div>
        <div>
          <div
            className="MuiGrid-root makeStyles-firstContainer-2 MuiGrid-container MuiGrid-item"
            style={
              Object {
                "padding": "5px",
              }
            }
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-2"
            >
              <img
                alt="Garen"
                className="makeStyles-img-1"
                src="http://ddragon.leagueoflegends.com/cdn/11.9.1/img/champion/Garen.png"
              />
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-4"
            >
              <h5
                className="MuiTypography-root makeStyles-championText-7 MuiTypography-h5"
              >
                Garen
              </h5>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                Avg. CS 178.6
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root makeStyles-kdaWinrateText-9 MuiTypography-body1"
              >
                8:1 KDA
              </p>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                7/3/17
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root makeStyles-kdaWinrateText-9 MuiTypography-body1"
              >
                58%
              </p>
              <p
                className="MuiTypography-root makeStyles-bottomText-8 MuiTypography-body1"
              >
                137 games
              </p>
            </div>
          </div>
          <hr
            className="MuiDivider-root makeStyles-divider-3"
          />
        </div>
      </div>
      <div
        hidden={true}
        id="tabpanel-1"
        role="tabpanel"
        style={
          Object {
            "width": "100%",
          }
        }
      />
      <div
        hidden={true}
        id="tabpanel-2"
        role="tabpanel"
        style={
          Object {
            "width": "100%",
          }
        }
      />
    </div>
  </div>,
]
`;
